src/indexer/utils/bitcoin/index.ts

```diff
@@ -0,0 +1,255 @@
+    amountInUSD = await coinMakerCapService.getValueInUSD(decodedDeposit.amount.toString(), decodedDeposit.resource.symbol)
+  } catch (error) {
+    logger.error((error as Error).message)
+    amountInUSD = 0
+  }
+
+  let transfer = await transferRepository.findTransfer(Number(depositNonce), originDomainId, destinationDomainId)
+
+  if (!transfer) {
+    transfer = await transferRepository.insertDepositTransfer({
+      amount: decodedDeposit.amount.toString(),
+      depositNonce: depositNonce,
+      destination: destinationAddress,
+      fromDomainId: originDomainId.toString(),
+      resourceID: decodedDeposit.resource.resourceId,
+      sender: "",
+      toDomainId: destinationDomainId.toString(),
+      usdValue: amountInUSD,
+    })
+  } else {
+    await transferRepository.updateTransfer(
+      {
+        amount: decodedDeposit.amount.toString(),
+        depositNonce: depositNonce,
+        destination: destinationAddress,
+        fromDomainId: originDomainId.toString(),
+        resourceID: decodedDeposit.resource.resourceId,
+        sender: "",
+        toDomainId: destinationDomainId.toString(),
+        usdValue: amountInUSD,
+      },
+      transfer.id,
+    )
+  }
+
+  const deposit = {
+    id: new ObjectId().toString(),
+    type: BitcoinTypeTransfer.Fungible,
+    txHash: txId,
+    blockNumber: blockHeight.toString(),
+    depositData: "",
+    timestamp: new Date(blockTime * 1000),
+    handlerResponse: "",
+    transferId: transfer.id,
+  }
+  await depositRepository.insertDeposit(deposit)
+
+  const feeData = {
+    id: new ObjectId().toString(),
+    transferId: transfer.id,
+    tokenSymbol: decodedDeposit.resource.symbol,
+    decimals: decodedDeposit.resource.decimals,
+    tokenAddress: decodedDeposit.resource.address,
+    amount: decodedDeposit.feeAmount.toString(),
+  }
+  await feeRepository.insertFee(feeData)
+}
+
+async function decodeExecutionEvent(tx: Transaction, resources: BitcoinResource[], client: RPCClient): Promise<DecodedExecution[] | undefined> {
+  let executionData: DecodedExecution[] | undefined
+
+  for (const vout of tx.vout) {
+    if (vout.scriptPubKey.type == OP_RETURN) {
+      // Extract OP_RETURN data (excluding OP_RETURN prefix)
+      const opReturnData = Buffer.from(vout.scriptPubKey.hex, "hex")
+      const data = opReturnData.subarray(2).toString()
+
+      // Data is in format syg_<hash>
+      const dataElems = data.split("_")
+      if (dataElems[0] == "syg") {
+        for (const vin of tx.vin) {
+          const senderTx = (await client.getrawtransaction({ txid: vin.txid, verbose: true })) as Transaction
+          const resourceFound = resources.find(resource => resource.address === senderTx.vout.at(vin.vout)?.scriptPubKey.address)
+          if (resourceFound) {
+            executionData = await fetchMetadata(dataElems[1])
+            break
+          }
+        }
+      }
+      break
+    }
+  }
+  return executionData
+}
+
+async function saveProposalExecution(
```

Comment:
A lot of this save functions have basically the same logic as other networks.
Would be prudent to check if it can be unified.

Reply:
I think it's better to do it in a separate PR. Along with refactoring `saveDeposit` functions.